docs/control-structures.rst

```diff
@@ -117,14 +127,14 @@ External Function Calls
 The expressions ``this.g(8);`` and ``c.g(2);`` (where ``c`` is a contract
 instance) are also valid function calls, but this time, the function
 will be called "externally", via a message call and not directly via jumps.
-Please note that function calls on ``this`` cannot be used in the constructor, as the
-actual contract has not been created yet.
+
+..note::
+  While it's possible to use ``address(this)`` to retrieve the address of the contract in the constructor, as the actual contract has not been created yet, you should not use ``this`` for anything else.
```

Comment:
You should especially be careful when using `this` in functions called from the constructor or when passing `this` to other contracts during the construction phase, as calls back into the contract will not work.